### PR TITLE
Twig itemdevice

### PR DESCRIPTION
**Pull request by stonebuzz**
Use twig for ```Item_Device``` (usefull for locked field)

Add ```Lock```.

remove useless ```fields``` from ```getSpecificities``` already handle by ```generic_show_form``` 

- ```locations_id```
- ```states_id```
- ```otherserial```
- ```serial```
- ```users_id```
- ```groups_id```.

Example : 

Simcard before :
image

Simcard after :
image

Item_DeviceMemory before :  
image

Item_DeviceMemory after :  
image

Note : 
The known fields of GLPI are displayed before those of the class.  
The order of the fields is therefore not the same.
I don't know if it is possible to do anything

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by cedric-anne — CHANGES_REQUESTED** [review]
`getSpecificities` is used in many places, inclunding in the logic that build seearch options, you cannot remove fields from there.

**Comment by stonebuzz** [comment]
> `getSpecificities` is used in many places, inclunding in the logic that build seearch options, you cannot remove fields from there.

agree with you

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Comment by stonebuzz** [comment]
sorry for that last commit 
Trying to get order as closest to the original
So i need drop generic_show_form to take control

**Review by trasher — APPROVED** [review]

File: templates/components/form/item_device.html.twig [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+      </div> {# .flex-row #}
+   </div> {# .flex-column #}
+</div> {# .flex-wrap #}
+{% if item_type == 'Contract' %}
+   {{ include('components/form/support_hours.html.twig') }}
+{% endif %}
+{{ include('components/form/buttons.html.twig') }}
+{% if no_inventory_footer is not defined or no_inventory_footer == false %}
+   {{ include('components/form/inventory_info.html.twig') }}
+{% endif %}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:236]: done

File: templates/components/form/item_device.html.twig [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+{% set field_options = {
+      'locked_fields': item.getLockedFields(),
+} %}
+
+<div class="card-body d-flex flex-wrap">
+   <div class="col-12 col-xxl-{{ item_has_pictures ? '9' : '12' }} flex-column">
+      <div class="d-flex flex-row flex-wrap flex-xl-nowrap">
+         <div class="row flex-row align-items-start flex-grow-1">
+            <div class="row flex-row">
+               {% block more_fields %}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:48]: done

File: templates/components/form/item_device.html.twig [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+ #
+ # You should have received a copy of the GNU General Public License
+ # along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ #
+ # ---------------------------------------------------------------------
+ #}
+
+{% import 'components/form/fields_macros.html.twig' as fields %}
+{% set params  = params ?? [] %}
+{% set no_header = true %}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:36]: done

File: templates/components/form/item_device.html.twig [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+{% set no_inventory_footer = true %}
+{% set item_has_pictures = item.hasItemtypeOrModelPictures() %}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:38]: done

File: templates/components/form/item_device.html.twig [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+                  {% endif %}
+
+
+                  {{ fields.autoNameField(
+                     'name',
+                     item,
+                     (item_type == 'Contact' ? __('Surname') : __('Name')),
+                     withtemplate,
+                     field_options
+                  ) }}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:116]: indeed

File: templates/components/form/item_device.html.twig [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                        field_options
+                     ) }}
+                  {% endif %}
+
+                  {% if item.isField('otherserial') %}
+                     {{ fields.autoNameField(
+                        'otherserial',
+                        item,
+                        __('Inventory number'),
+                        withtemplate,
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:168]: done

File: templates/components/form/item_device.html.twig [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+                     {{ fields.dropdownField(
+                        'Group',
+                        'groups_id',
+                        item.fields['groups_id'],
+                        'Group'|itemtype_name,
+                        field_options|merge({
+                           'entity': item.fields['entities_id'],
+                           'condition': {'is_itemgroup': 1},
+                        })
+                     ) }}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:222]: Exist in ```glpi_items_devicesimcards``

File: templates/components/form/item_device.html.twig [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+                  {% endif %}
+               {% endblock %}
+            </div> {# .row #}
+         </div> {# .row #}
+      </div> {# .flex-row #}
+   </div> {# .flex-column #}
+</div> {# .flex-wrap #}
+{% if item_type == 'Contract' %}
+   {{ include('components/form/support_hours.html.twig') }}
+{% endif %}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig:232]: done

File: templates/components/form/item_device.html.twig [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+{{ include('components/form/buttons.html.twig') }}
+{% if no_inventory_footer is not defined or no_inventory_footer == false %}
+   {{ include('components/form/inventory_info.html.twig') }}
+{% endif %}
+
+{% if params['formfooter'] is null or params['formfooter'] == true %}
```
[the remark this answers is not in the document]
Comment (stonebuzz) [reply templates/components/form/item_device.html.twig]: agree formfooter and date not needed here

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**